Documentation work order (#37)
landinfotech/mycivitas: docs/src/user/manual/work-order.md

# Work Order Page User Manual

Welcome to the `Work Order` page, a central hub for managing and tracking work orders. This user manual will guide the user through the features available on this page.

![work order](./img/work-order-1.png)

1. **Work Order:** Click on the `Work Order` option available in the navigation menu. Upon clicking, the user will be redirected to the Work Order page. By default, the Work Order table will be displayed. The table provides an overview of all existing tickets, including key details such as status, assignee, and due date.

## Work Order Page

![work order page](./img/work-order-2.png)

### Table Section

1. **Table:** The table section displays the list of tickets. Each row represents a ticket. It includes columns such as `Ticket`, `Type` `Priority`, `Status`, `Created`, `Operator`, `Submitter`, `Start Date`, and `Due Date`. Each column has an up/down arrow to indicate the option to display the list in ascending or descending order.

> 2. **Show Entries Dropdown:** Utilize the `Show Entries` dropdown to choose the number of entries as per user convenience to display on the page. Options include 10, 25, 50, and 100 entries, allowing for customized viewing.

Reviewer: ```suggestion
2. **Show Entries Dropdown:** Utilise the `Show Entries` dropdown to choose the number of entries as per user convenience to display on the page. Options include 10, 25, 50, and 100 entries, allowing for customised viewing.
```

    ![10 entries](./img/work-order-3.png)

3. **Search Option:** Use the search functionality to find a specific ticket efficiently. Insert relevant keywords in the search bar to narrow down the search results.

    ![searched result](./img/work-order-4.png)

4. **Ticket Name:** Click on the ticket name within the table to:

    - View detailed information about the ticket.
    - Edit the ticket details.
    - Delete the ticket if needed.

    Upon clicking, the user will be redirected to the `Work Order` Edit page for further actions.

    ![edit work order](./img/work-order-5.png)

    Welcome to the `Work Order Edit` page, where the user can fine-tune and manage the details of their work orders.

    1. **Location on the Map:** This section displays the specific location on the map for which the work order ticket is created. Gain visual insights into the geographical context of the work order.

>     2. **Plus/Minus Icons:** Utilize the `Plus/Minus` icons for zooming in and out on the map. Adjust the map view to focus on specific details or get a broader perspective.

Reviewer: ```suggestion
    2. **Plus/Minus Icons:** Utilise the `Plus/Minus` icons for zooming in and out on the map. Adjust the map view to focus on specific details or get a broader perspective.
```

        ![edit ticket form](./img/work-order-6.png)
        
        The user can update the following fields as needed:
        - Title
        - Description
        - Priority (dropdown selection)
        - Start Date
        - Due Date
        - Operator (dropdown selection)
        - Expected Time to Complete Task (hours and minutes)

        1. **Cancel Button:** Use the `Cancel` button to discard any changes made in the Edit Ticket Form. Ideal for reverting back to the original ticket details without saving modifications.

        2. **Save Button:** Click the `Save` button to confirm and save the updated information. Ensure that the changes are recorded and reflected in the work order details.

    7. **Delete:** The user can delete the ticket using the `Delete` button. When the user clicks on the button, a confirmation popup opens, ensuring the user's intention before proceeding with the deletion of the ticket.

5. **Previous/Next Button:** Navigate through multiple pages of the `Work Order` table using the `Previous` and `Next` buttons. Easily access different sets of work orders based on the user's viewing preferences.

> ### TimeLine Section

Reviewer: ```suggestion
### Timeline Section
```

Explore the `Timeline` section, a central hub for navigating through various tickets and their details. Here's the guide to efficiently use this feature.

![timeline section](./img/work-order-8.png)

> 1. **TimeLine:** The `Timeline` shows all activities related to a specific Work Order. In the middle of the page, the user will find the title of the ticket along with its description. The created date of the ticket is also prominently displayed.

Reviewer: ```suggestion
1. **Timeline:** The `Timeline` shows all activities related to a specific Work Order. In the middle of the page, the user will find the title of the ticket along with its description. The created date of the ticket is also prominently displayed.
```

2. **View Ticket:** A `View Ticket` link is available, providing a quick way to access the full details of the displayed ticket. When the user clicks on this link they will redirected to the edit ticket page.

    ![ticket edit page](./img/work-order-9.png)

> 3. **Useful feature:** The user can use these features for better visualization.

Reviewer: ```suggestion
3. **Useful feature:** The user can use these features for better visualisation.
```

        ![ticket edit page](./img/work-order-12.png)

4. **Status:** This shows the status of the ticket and the assigned person's email.

### Reports & Statistics Section

Unlock valuable insights into ticket management with the Reports & Statistics section. Here's a comprehensive guide to navigating and interpreting the presented information.

![reports & statistics](./img/work-order-13.png)

1. **Report & Statistics:** This section provides the average number of days until a ticket is closed for all tickets. Additionally, it offers insights into the average number of days until ticket closure specifically for tickets opened in the last 60 days. Click `here` to delve into more granular data.

    - **Queue (Clickable):** Click on the queue type to explore detailed information about tickets in that specific category.

    - **New:** Number of tickets in the `New` status.

    - **Open:** Number of tickets in the `Open` status.

    - **Resolved:** Number of tickets in the `Resolved` status.

    - **Closed:** Number of tickets in the `Closed` status.

    - **Time Spent:** Insights into the cumulative time spent on all tickets.

> 2. **Reports By User:**  Explore detailed insights into ticket management by utilizing the `Reports by User` section. Choose from four distinct options to view both table and chart representations.

Reviewer: ```suggestion
2. **Reports By User:**  Explore detailed insights into ticket management by utilising the `Reports by User` section. Choose from four distinct options to view both table and chart representations.
```

>     1. **By Priority:** Click on `By Priority` to access a comprehensive table and chart showcasing ticket distribution based on priority. Understand the distribution of tickets categorized by their priority levels.

Reviewer: ```suggestion
    1. **By Priority:** Click on `By Priority` to access a comprehensive table and chart showcasing ticket distribution based on priority. Understand the distribution of tickets categorised by their priority levels.
```
